### PR TITLE
QSP-7 Gas Usage / for Loop Concerns - Swap.cancel()

### PR DESCRIPTION
**Pull request by ejwessel**
## Description
https://github.com/airswap/airswap-protocols/issues/294
Update the `Swap.cancel()` to properly indicate issues with gas usage and how to remedy.

- [ ] N/A
- [ ] New features
- [ ] Bug fixes
- [X] Typo/small tweaks

## Changes
Update @dev with description of gas usage.

## Tests
N/A

### CODE REVIEW

File: source/swap/contracts/Swap.sol [hunk 1]
```diff
@@ -172,6 +172,10 @@ contract Swap is ISwap {
     * @notice Cancel one or more open orders by nonce
     * @dev Cancelled nonces are marked UNAVAILABLE (0x01)
     * @dev Emits a Cancel event
+    * @dev If the list of nonces to cancel is too long, 
+      there is a potential issue of out of gas error.
```
Comment (alicevhenshaw) [inline source/swap/contracts/Swap.sol]: Add a star at the beginning of each line

**Comment by alicevhenshaw** [comment]
I personally don't think we need to explain ourselves here.

My maths on `Swap.cancel` having analysed cancelling different length arrays of nonces:
- Function overhead: 22201
- Cost to cancel a nonce: 22831

Therefore cost to cancel 100 nonces: 2,305,301
Total nonces you could cancel in 1 transaction on mainnet before hitting the block gas limit: 437

If someone is trying to cancel even 100 nonces I’d be surprised, let alone 437. They can use the function `invalidate` to set a minimum nonce.

**Comment by ejwessel** [comment]
I agree with you that it may be an overly informative to explain ourselves. However, do you think it reflects better on us to denote that we're aware about the limitations and have vocalized it? I'm happy to close this either way.

**Comment by alicevhenshaw** [comment]
Currently it's a long explanation that would make people worry it actually might happen to them, without saying what the limits of the function really are. Imo if its going to be included you should keep it brief and include actual data - I'd rather have a 1 liner that just says:
`* @dev Out of gas may occur on arrays of length > 400`
This way the reader gets more data on when the error occurs.

**Comment by ejwessel** [comment]
I'm all for keeping a concise message. 

I'll wait for input from others on where we should go from here.

**Comment by poiresel** [comment]
* @dev Out of gas may occur on arrays of length > 400 seems like a nice informative succinct message.

File: source/swap/contracts/Swap.sol [hunk 2]
```diff
@@ -172,6 +172,7 @@ contract Swap is ISwap {
     * @notice Cancel one or more open orders by nonce
     * @dev Cancelled nonces are marked UNAVAILABLE (0x01)
     * @dev Emits a Cancel event
+    * @dev Out of gas may occur on arrays of length > 400
```
Comment (poiresel) [inline source/swap/contracts/Swap.sol]: occur in arrays versus occur on arrays

**Pull request merged**